host-controller/src/main/java/org/jboss/as/host/controller/ServerInventory.java

```diff
@@ -171,14 +175,32 @@
     /**
      * Re-establishes management communications with a server following a restart of the Host Controller process.
      *
+     * <p>The servers, which are still stopping, can be requested to be restarted using {@code requestRestart} parameter.
+     * This restart operation will be done in background once the servers are completely stopped, a new {@code authKey} will
+     * be generate for the delayed start. Similarly, the {@code domainModel} used in the delayed start will be the one that exits
+     * at that moment.
+     *
+     * <p> The servers, which are already started or stopped, are not affected by this parameter.
+     * For those cases, use {@link #startServer(String, ModelNode)} or any overloaded method.
+     *
+     * <p> The delayed start will be done always acquiring the Host Controller lock, the {@code restartBlocking} param on this method, can
+     * be used to limit the time of that lock, if it is {@code true}, then the lock will be released when the server is
+     * in STARTED state, otherwise when the server is in STARTING state.
+     *
      * @param serverName the name of the server
      * @param domainModel the configuration model for the domain
      * @param authKey the authentication key
      * @param running whether the process was running. If {@code false}, the existence of the server will be
      *                recorded but no attempt to contact it will be made
      * @param stopping whether the process is currently stopping
+     * @param blockUntilStopped wait until the server is stopped
+     * @param requestRestart whether the server will be restarted in background once it is stopped
+     * @param restartBlocking whether the server will be restarted in background in a thread waiting until the server is started, otherwise the thread
+     *                        will wait until the server is starting.
+     *
+     * @return Returns {@code true} if the server was selected to be restarted, otherwise it returns false.
      */
-    void reconnectServer(String serverName, ModelNode domainModel, String authKey, boolean running, boolean stopping);
+    boolean reconnectServer(String serverName, ModelNode domainModel, String authKey, boolean running, boolean stopping, boolean blockUntilStopped, boolean requestRestart, boolean restartBlocking);
```

Comment:
I don't see any code using this return value.

Reply:
You're right, initially, I thought it could be useful to the caller to know if the server was elected to be restarted or not, and indeed I was using it in a previous version, but later it was not longer necessary. Removed.